tests/test-security.php

```diff
@@ -118,17 +138,129 @@ public function test__lost_password_limit() {
 
 	}
 
-	public function setUp(): void {
-		parent::setUp();
+	public function test__wpcom_vip_track_auth_attempt__defaults() {
+		wpcom_vip_track_auth_attempt( $this->test_username, CACHE_GROUP_LOGIN_LIMIT );
 
-		// phpcs:ignore WordPress.Security.NonceVerification.Missing
-		$this->original_post = $_POST;
+		$username_count = wp_cache_get( $this->test_username, CACHE_GROUP_LOGIN_LIMIT );
+		$this->assertSame( 1, $username_count );
+
+		$ip_count = wp_cache_get( $this->test_ip, CACHE_GROUP_LOGIN_LIMIT );
+		$this->assertSame( 1, $ip_count );
+
+		$ip_username_count = wp_cache_get( $this->test_ip . '|' . $this->test_username, CACHE_GROUP_LOGIN_LIMIT );
+		$this->assertSame( 1, $ip_username_count );
 	}
 
-	public function tearDown(): void {
-		$_POST = $this->original_post;
+	public function test__wpcom_vip_login_limiter_on_success__decrease_count() {
+		$original_count  = 5;
+		$decreased_count = $original_count - 1;
+		wp_cache_set( $this->test_username, $original_count, CACHE_GROUP_LOGIN_LIMIT );
+		wp_cache_set( $this->test_ip, $original_count, CACHE_GROUP_LOGIN_LIMIT );
+		wp_cache_set( $this->test_ip . '|' . $this->test_username, $original_count, CACHE_GROUP_LOGIN_LIMIT );
 
-		parent::tearDown();
+		wpcom_vip_login_limiter_on_success( $this->test_username );
+
+		$username_count = wp_cache_get( $this->test_username, CACHE_GROUP_LOGIN_LIMIT );
+		$this->assertSame( $original_count, $username_count ); // username is NOT decreased
+
+		$ip_count = wp_cache_get( $this->test_ip, CACHE_GROUP_LOGIN_LIMIT );
```

Comment:
Flaky how? 

Does the tests architecture mock the object cache (with, an in-memory cache, say), since we don't want to be testing the actual functionality of wp_cache_set() and wp_cache_get() themselves, only that they've been called with the right arguments. This is where BrainMonkey could also help.

Reply:
If it's flaky per PHP / PHPUnit version, then [Yoast's WPTestUtils](https://github.com/Yoast/wp-test-utils) (or at least [PHPUnit-Polyfills](https://packagist.org/packages/yoast/phpunit-polyfills)) can help, and that requires using a slightly different version of `tearDown()` method name that works across different PHPUnit versions.

Reply:
Oh I tracked it down. It was due to the sanitization of usernames somehow working differently on multisite (lowercasing text).